### DISCUSSION TITLE
ability to dump entire state of DUT for reloading

### DISCUSSION THREAD

**Discussion by gaclee3b** [cocotb/cocotb#3156] [Proposals]
I've been running some simulations that require a good amount of time for my design to go through an initialisation / start up procedure. The procedure has to be repeated every time until I can get to the tests I'm trying to run. 

I would like the ability to bypass the need to rerun the initialisation period of the simulation for subsequent tests by allowing the DUT to be primed into that post-initialised state before the simulation starts. 

Is there a way for cocotb to dump the state of a DUT for a specified point in the simulation that can be pre-loaded into the DUT before a simulation starts?

**Comment by cmarqu** [comment]
@mciepluc had some code for doing this, but I cannot quickly find it again.

**Comment by gaclee3b** [comment]
has there been any updates on this, or any plans to officially add this as a feature?

**Reply by imphil** [reply]
I don't know of anyone who has made progress on this. If you'd like to get such a feature added to cocotb, please open a pull request with an implementation, we're happy to review it!

**Comment by themperek** [comment]
This was working at some point: https://github.com/themperek/apbi2c_cocotb_example/blob/parallel/tb/checkpoint.py
